distributed/protocol/serialize.py

```diff
@@ -124,16 +137,26 @@ def serialize(x):
     if isinstance(x, Serialized):
         return x.header, x.frames
 
-    typ = type(x)
-    name = typename(typ)
-    if name in serializers:
-        header, frames = serializers[name](x)
-        header['type'] = name
-    else:
-        if _find_lazy_registration(name):
-            return serialize(x)  # recurse
-        header, frames = {}, [pickle.dumps(x)]
-
+    serializers = serializers or ['dask', 'pickle']  # maybe a global option
+    header = None
+    for serializer in serializers:
+        if serializer == 'dask':
+            typ = type(x)
+            name = typename(typ)
+            if name in class_serializers:
+                header, frames = class_serializers[name](x)
+                header['type'] = name
+            elif _find_lazy_registration(name):
+                return serialize(x, serializers)  # recurse
+        else:
+            try:
+                header, frames = {'type': serializer}, [deser[serializer][0](x)]
+            except:
+                # this serializer failed somehow - try next one
+                # TODO: consider logging this event
+                continue
+    if header is None:
+        raise TypeError('Failed to serialize %s with %s' % (x, serializers))
```

Comment:
I would actually prefer to pass back something specific to the other side of the comm and only raise there - this way you get an exception that shows up in the worker log, and a very long exception log in the client. 
Actually raising the real exception may be dangerous, since this traceback is pickled, right?

Reply:
Would it be reasonable to pass header='error', frames=bytes(error message)?

Reply:
I agree that the other side of the comm should be aware of the serialization error.  It may be that there are already mechanisms to handle this (I'm not sure though).  It is useful to push the exception and traceback back to the user as far as possible.  This information helps them to understand what went wrong and helps them (or us) to diagnose the issue.  We often get issues today that have tracebacks with information like "Lock can not be serialized".  It is important that we continue to get this information and that it doesn't get swallowed by an except somewhere.

Reply:
Can it be 100% guaranteed that pickling the exception and traceback are safe? We of course raise the exception in this case, so we can be sure not to include the object that failed to serialise as part of the exception instance.